### ISSUE TITLE
No body for gopherjs serve

### ISSUE THREAD

**Issue by EtienneBruines** [gopherjs/gopherjs#451]
## TL;DR
Programmatically adding a `<body>`, results in two `<body>`-tags in Google Chrome. 

## Longer version
I've noticed that the files generated by `gopherjs seve` are rather empty. 

```html
<html><head><meta charset="utf-8"><script src="pong.js"></script></head></html>
```

`dom.GetWindow().Document().(dom.HTMLDocument).Body()` returns `nil`, forcing us to create a `<body>` tag ourselves (programmatically). However, Google Chrome tends to do the same (I guess). 

We used this code to generate a new `<body>`:

```go
js.Global.Get("document").Set("body", js.Global.Get("document").Call("createElement", "body"))
```

![](https://www.huperwebs.nl/upload/screenshot-2016-05-05_19:05:23.jpg)

Are we doing something wrong here? Or is there a reason why the `gopherjs` file doesn't include a `<body>` tag?

-------

We tend to check for "does the body exist" very early in the loop. Possible workarounds: 
* Adding  `defer` to the `<script>` in a html file (which we had to create, in order to do this)
* Moving the `<script>` tag to within the html body (for which we needed to create a file). 

I was wondering if there was a way to do this *without* creating an HTML file.

**Comment by neelance** [comment]
This was done to prevent flickering on page load (rendering an empty body). You are right that now there are two bodies. This seems to be a recent change to Chrome. Needs further investigation...

**Comment by neelance** [comment]
I've added the body tag. Also take a look at https://github.com/gopherjs/vecty/pull/43.

**Comment by EtienneBruines** [comment]
Thank you for the effort. 

Wouldn't adding the `<script>` tag to the `<body>` solve the problem as well?

Because even though https://github.com/gopherjs/vecty/pull/43/files looks nice, it doesn't really work when you need the function to do stuff before returning.

**Comment by neelance** [comment]
If you add `<script>` to the `<body>`, then it's flickering because the empty body gets rendered first.
